pymongo/pool.py

```diff
@@ -1289,6 +1335,15 @@ def get_socket(self, all_credentials, checkout=False):
             if not checkout:
                 self.return_socket(sock_info)
 
+    def _raise_if_not_ready(self):
+        if self.opts.pause_enabled and self.state == PAUSED:
```

Comment:
Any reason we don't simply check `self.state != READY`? Seems like we are leaving out the possibility of the state being CLOSED?